content/articles/react-components-instantiation/index.md

```diff
@@ -0,0 +1,192 @@
+    <h1>Welcome to React World</h1>,
+   document.getElementById('root')
+  );
+```
+It would be best to give it the same name as the component class. For example, let's now create a component instance by adding <Valley/> which will be our component's class below our code, as shown below:
+
+```js
+ import React from 'react';
+ import ReactDOM from 'react-dom';
+
+   class Valley extends React.Component {
+    render() {
+    return <h1>Welcome to React world</h1>;
+   }
+ }
+   <Valley />
+```
+The code appears this way after creating a component instance. So there’s your component instance! It gets instantiated when class `Valley` gets passed as an argument in `ReactDOM.render`. 
+
+Let's now render our component:
+```js
+    ReactDOM.render(
+  <Valley />,
+  document.getElementById('app')
+  );
+```
+
+The `render` method is called when `ReactDOM.render()` tells our component class `<Valley />` to call its `render` method. Class `<Valley />` then calls its `render` method, which  returns the JSX element `<h1>Welcome to React World</h1>`.  This will enable “Welcome to React World” to be displayed on the screen.
+
+#### Using a new keyword 
+The `new` keyword creates a blank, simple JavaScript item. Its syntax is `new constructor[([arguments])]`. When you create an object, a class instance is created.The class name to be instantiated is provided by the constructor name. The constructor initializes the new object.A reference is returned to the object created by the `new` operator.
+
+The `new` keyword makes the `this` keyword point to the object that has been newly created. Each time `this` is mentioned, the `new` keyword executes the constructor function, using the object that has been newly created. Hence, the newly created object is returned.
+
+ #### Instantiating function components
+Function components are simple JavaScript components. They don't use a `render` method. Instead, they are responsible for rendering the user interface. They also accept and use props. A function component is considered a "dump" because it simply accepts data.
+
+ ***example***
+ ```js
+ function Student (marks) {
+  this.marks = marks;
+  }
+
+ const study = new Student(50);
+
+  console.log(study);
+  ```
+The code above outputs the student's marks. This is because the JS engine will see the `new` keyword, then it will create a new object and set the prototype  to `Student.prototype`. If a property is undefined in the new object that is requested, the script will check the [[Prototype]] object for the property.
+
+Let's add `console.log(Object.getPrototypeOf(studdy) === Student.prototype)` to see if prototype of `study` is `Student` which will output true. If we add `Student` instead of `study` as shown `console.log(Object.getPrototypeOf(Student) === Student.prototype)` it will output false because Student is not a prototype of Student.
+
+****What happens in the code above?****
+- `const studdy` means: Memory is needed for variable declaration.
```

Comment:
What about this?

Reply:
In this code am showing the reader how to find the prototype of  the new object. object's prototype property is invisible.So I used Object.getPrototypeOf(obj) method to access prototype object.